app-android/src/main/java/com/ustadmobile/port/android/view/ReportFilterEditFragment.kt

```diff
@@ -272,4 +294,173 @@ class ReportFilterEditFragment : UstadEditFragment<ReportFilter>(), ReportFilter
+                )
+
+                 Spacer(modifier = Modifier.width(10.dp))
+
+                UstadMessageIdOptionExposedDropDownMenuField(
+                    modifier = Modifier.weight(0.5F),
+                    value = uiState.reportFilter?.reportFilterDropDownValue ?: 0,
+                    label = stringResource(R.string.report_filter_edit_values),
+                    options = ContentCompletionStatusConstants.CONTENT_COMPLETION_STATUS_MESSAGE_IDS,
+                    error = uiState.valuesError,
+                    enabled = uiState.fieldsEnabled,
+                    onOptionSelected = {
+                        onReportFilterChanged(uiState.reportFilter?.shallowCopy{
+                            reportFilterDropDownValue = it.value
+                        })
+                    },
+                )
+            }
+        }
+
+        item {
+            UstadTextEditField(
+                value = uiState.reportFilter?.reportFilterValue ?: "",
+                label = stringResource(id = R.string.report_filter_edit_values),
+                error = uiState.valuesError,
+                enabled = uiState.fieldsEnabled,
+                keyboardOptions = KeyboardOptions(keyboardType = KeyboardType.Number),
+                onValueChange = {
+                    onReportFilterChanged(uiState.reportFilter?.shallowCopy{
+                        reportFilterValue = it
+                    })
+                }
+            )
+        }
+
+        item {
+            Row {
+                UstadTextEditField(
+                    modifier = Modifier.weight(0.5F),
+                    value = uiState.reportFilter?.reportFilterValueBetweenX ?: "",
+                    label = stringResource(id = R.string.from),
+                    error = uiState.valuesError,
+                    enabled = uiState.fieldsEnabled,
+                    keyboardOptions = KeyboardOptions(keyboardType = KeyboardType.Number),
+                    onValueChange = {
+                        onReportFilterChanged(uiState.reportFilter?.shallowCopy{
+                            reportFilterValueBetweenX = it
+                        })
+                    }
+                )
+
+                Spacer(modifier = Modifier.width(10.dp))
+
+                UstadTextEditField(
+                    modifier = Modifier.weight(0.5F),
+                    value = uiState.reportFilter?.reportFilterValueBetweenY ?: "",
+                    label = stringResource(id = R.string.toC),
+                    error = uiState.valuesError,
+                    enabled = uiState.fieldsEnabled,
+                    keyboardOptions = KeyboardOptions(keyboardType = KeyboardType.Number),
+                    onValueChange = {
+                        onReportFilterChanged(uiState.reportFilter?.shallowCopy{
+                            reportFilterValueBetweenY = it
+                        })
+                    }
+                )
+            }
+        }
+
+        items(
```

Comment:
Please add reportFilterUidAndLabelListVisible boolean to UiState. This should only be visible of that is true.